### PR TITLE
deps/proc log

### PR DESCRIPTION
**Pull request by wraithgar**
- **deps: proc-log@4.0.0**
- **feat!: output using proc-log**

### CODE REVIEW

File: lib/run-script-pkg.js [hunk 1]
```diff
@@ -59,10 +45,17 @@ const runScriptPkg = async options => {
...
+  output.standard('')
+  if (pkg._id) {
+    output.standard(`> ${pkg._id} ${event}`)
+  } else {
+    output.standard(`> ${event}`)
+  }
+  if (args.length) {
+    output.standard(`> ${cmd.trim().replace(/\n/g, '\n> ')} ${args.join(' ')}`)
   }
+  output.standard('')
```
Comment (lukekarrys) [inline lib/run-script-pkg.js:58]: I think this should still build a multiline string and call `output.standard()` once.

**Review by lukekarrys — COMMENTED** [review]
We can remove all mentions of `banner` from the README.

File: test/run-script-pkg.js [hunk 2]
```diff
@@ -6,18 +6,17 @@ const isWindows = process.platform === 'win32'
...
-  t.teardown(() => console.log = consoleLog)
-  return logs
-}
+const output = []
+process.on('output', (level, ...args) => {
+  if (level === 'standard') {
+    output.push([...args])
+  }
+})
+t.afterEach(() => output.length = 0)
```
Comment (lukekarrys) [inline test/run-script-pkg.js:16]: We should remove the `output` event from `process` on teardown I think. Won't change anything currently but could prevent a footgun in future refactors.

File: lib/run-script-pkg.js [hunk 3]
```diff
@@ -59,10 +45,17 @@ const runScriptPkg = async options => {
     return { code: 0, signal: null }
   }
 
-  if (stdio === 'inherit' && banner !== false) {
-    // we're dumping to the parent's stdout, so print the banner
-    console.log(bruce(pkg._id, event, cmd, args))
+  let banner
```
Comment (lukekarrys) [inline lib/run-script-pkg.js:48]: I think we still might want to guard this with `if (stdio === 'inherit')`.

For example, we set `stdio: pipe` if `--foreground-scripts=false` is set. In that case we will still get `output` events in the display layer but we'll need to know if they came from a child process so we can use `config.foregroundScripts` to hide them.

File: test/run-script-pkg.js [hunk 4]
```diff
@@ -33,34 +34,11 @@ t.test('run-script-pkg', async t => {
         scripts: {},
       },
     })
-    t.strictSame(logs, [['\n> foo@1.2.3 foo\n> bar\n> baz\n']])
+    t.strictSame(output, [['\n> foo@1.2.3 foo\n']])
```
Comment (wraithgar) [inline test/run-script-pkg.js:37]: This looks wrong

**Review by lukekarrys — APPROVED** [review]

**Review by lukekarrys — APPROVED** [review]

**Review by nishantms — APPROVED** [review]

**Pull request merged**